core/src/sumeragi/main_loop.rs

```diff
@@ -785,6 +774,7 @@ fn reset_state(
     old_view_change_index: &mut u64,
     current_latest_block_height: u64,
     old_latest_block_height: &mut u64,
+    lastest_block: &VersionedCommittedBlock,
```

Comment:
if you send both latest block and voting block, aren't some of the arguments given to this fn redundant?